subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -267,11 +267,25 @@ uint8_t ll_adv_params_set(uint16_t interval, uint8_t adv_type,
 				return BT_HCI_ERR_UNSUPP_FEATURE_PARAM_VAL;
 			}
 
-			adv_type = leg_adv_type[evt_prop & 0x03];
+#if defined(CONFIG_BT_CTLR_ADV_PERIODIC)
+			/* disallow changing to legacy advertising while
+			 * periodic advertising enabled.
+			 */
+			if (adv->lll.sync) {
+				const struct ll_adv_sync_set *sync;
+
+				sync = HDR_LLL2ULL(adv->lll.sync);
+				if (sync->is_enabled) {
+					return BT_HCI_ERR_INVALID_PARAM;
+				}
+			}
+#endif /* CONFIG_BT_CTLR_ADV_PERIODIC */
 
 			/* high duty cycle directed */
 			if (evt_prop & BT_HCI_LE_ADV_PROP_HI_DC_CONN) {
 				adv_type = 0x01; /* PDU_ADV_TYPE_DIRECT_IND */
+			} else {
+				adv_type = leg_adv_type[evt_prop & 0x03];
 			}
```

Comment:
```suggestion
			/* high duty cycle directed */
			if (evt_prop & BT_HCI_LE_ADV_PROP_HI_DC_CONN) {
				adv_type = 0x01; /* Index of PDU_ADV_TYPE_DIRECT_IND in pdu_adv_type[] */
			} else {
				/* We take advantage of the fact that 2 LS bits of evt_prop can map to legacy advertising PDU types by using a look up */
				adv_type = leg_adv_type[evt_prop & BIT_MASK(sizeof(leg_adv_type)];
			}
```

0x03 is BIT_MASK(sizeof(leg_adv_type)) which is to restrict the index to the size of the lookup table.

Is the about comments and code suggestion acceptable?

Reply:
Hmm, that is actually making it more confusing to me. So `adv_type` is not an adv_type, but rather an index of a value in a local array that is unordered:
```c
			uint8_t const leg_adv_type[] = {
				0x03, /* PDU_ADV_TYPE_NONCONN_IND */
				0x04, /* PDU_ADV_TYPE_DIRECT_IND */
				0x02, /* PDU_ADV_TYPE_SCAN_IND */
				0x00  /* PDU_ADV_TYPE_ADV_IND */
};
```

but only when `evt_prop & BT_HCI_LE_ADV_PROP_HI_DC_CONN` else it is a value in the array? 

I think using the same variable as an index *and* as a value is hard to follow in terms of readability.

Reply:
`adv_type` is an index into `pdu_adv_type[]`. `leg_adv_type[]` is second level of lookup from evt_prop value that gives an index into `pdu_adv_type[]`. Do suggest changes that can make it clearer to understand the level of lookup.

Reply:
I see. That explains it. Would possibly also suggest to change `adv_type` to `pdu_adv_type_index` to make it very explicit that the variable is a index to the `pdu_adv_type` array

Reply:
`adv_type` is the Advertising Type parameter name in HCI LE Set Advertising Parameters command that is used in the lookup `pdu_adv_type[]`, will keep the name as-is.

Updated the PR will additional comments in code.